grakn-engine/src/test/java/ai/grakn/engine/rpc/GrpcServerTest.java

```diff
@@ -0,0 +1,578 @@
+import ai.grakn.rpc.generated.GraknGrpc.GraknStub;
+import ai.grakn.rpc.generated.GraknOuterClass;
+import ai.grakn.rpc.generated.GraknOuterClass.Commit;
+import ai.grakn.rpc.generated.GraknOuterClass.Done;
+import ai.grakn.rpc.generated.GraknOuterClass.ExecQuery;
+import ai.grakn.rpc.generated.GraknOuterClass.Infer;
+import ai.grakn.rpc.generated.GraknOuterClass.Next;
+import ai.grakn.rpc.generated.GraknOuterClass.Open;
+import ai.grakn.rpc.generated.GraknOuterClass.QueryResult;
+import ai.grakn.rpc.generated.GraknOuterClass.Stop;
+import ai.grakn.rpc.generated.GraknOuterClass.TxRequest;
+import ai.grakn.rpc.generated.GraknOuterClass.TxResponse;
+import ai.grakn.rpc.generated.GraknOuterClass.TxType;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static ai.grakn.engine.rpc.GrpcUtil.hasMessage;
+import static ai.grakn.engine.rpc.GrpcUtil.hasStatus;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Felix Chapman
+ */
+public class GrpcServerTest {
+
+    private static final String EXCEPTION_MESSAGE = "OH DEAR";
+    private static final GraknException EXCEPTION = new GraknException(EXCEPTION_MESSAGE) {
+        private static final long serialVersionUID = 8491972914457727509L;
+    };
+
+    private static final int PORT = 5555;
+    private static final String MYKS = "myks";
+    private static final String QUERY = "match $x isa person; get;";
+
+    private final EngineGraknTxFactory txFactory = mock(EngineGraknTxFactory.class);
+    private final GraknTx tx = mock(GraknTx.class);
+    private final GetQuery query = mock(GetQuery.class);
+
+    private GrpcServer server;
+
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    // TODO: usePlainText is not secure
+    private final ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", PORT).usePlaintext(true).build();
+    private final GraknStub stub = GraknGrpc.newStub(channel);
+
+    @Before
+    public void setUp() throws IOException {
+        server = GrpcServer.create(PORT, txFactory);
```

Comment:
This `server` is never used, do we need it?

Reply:
Even though it's not used in the tests, it opens up a gRPC server that the tests talk to.